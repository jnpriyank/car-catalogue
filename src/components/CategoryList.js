import React from "react";

const CategoryList = (props) => {

    const onClickHandler = (e) => {
        props.onCategoryChange(e.target.innerText);
    }

    return (
        <>
            {
                props.categories.map((category) => <div key={category.id} className={"category" + (category.name === props.selectedCategory? " selected": "")} onClick={onClickHandler}>{category.name}</div> )
            }
        </>
    );
};

export default CategoryList;