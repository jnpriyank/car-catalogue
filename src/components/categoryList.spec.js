import React from "react";
import { shallow } from "enzyme";
import CategoryList from "./categoryList";

describe("category list component", () => {
    it.only("should be rendered successfully, When categories are passed", () => {
        const categories = [
            {
                id: 1,
                name: "Hatchbacks",
                description: "Hatchbacks description"
            },
            {
                id: 2,
                name: "Vans",
                description: "Vans description"
            },
        ];
        const selectedCategory = "Vans";
        const onCategoryChangeHandler = jest.fn().mockImplementation((category) => {});

        const wrapper = shallow(<CategoryList 
                                    categories={categories} 
                                    selectedCategory={selectedCategory}
                                    onCategoryChange={onCategoryChangeHandler}
                                >
                                </CategoryList>);

        // assertions
        // should have two divs with class category
        expect(wrapper.find(".category")).toHaveLength(2);

        // second/last div should have text as 'Vans' and should have class selected
        expect(wrapper.find(".category.selected").text()).toBe("Vans");

        // simulate click on first item
        wrapper.find(".category").first().simulate("click", {
            target: {
                innerText: "Hatchbacks",
            },
        });

        // check the method inside handler is called
        expect(onCategoryChangeHandler).toHaveBeenCalled();
        expect(onCategoryChangeHandler).toHaveBeenCalledWith("Hatchbacks");

        expect(wrapper).toMatchSnapshot();
    });

    it("should not render anything, When categories are not passed", () => {
        const categories = [];
        const selectedCategory = "";
        const wrapper = shallow(<CategoryList categories={categories} selectedCategory={selectedCategory}></CategoryList>);

        expect(wrapper.find(".category")).toHaveLength(0);
        expect(wrapper).toMatchSnapshot();
    });
});