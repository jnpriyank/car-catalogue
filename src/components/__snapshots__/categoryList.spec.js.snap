// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`category list component should be rendered successfully, When categories are passed 1`] = `
<Fragment>
  <div
    className="category"
    key="1"
    onClick={[Function]}
  >
    Hatchbacks
  </div>
  <div
    className="category selected"
    key="2"
    onClick={[Function]}
  >
    Vans
  </div>
</Fragment>
`;

exports[`category list component should not render anything, When categories are not passed 1`] = `null`;
